filebeat/autodiscover/builder/hints/config.go

```diff
@@ -59,7 +59,12 @@ func (c *config) Unpack(from *common.Config) error {
 			}
 		} else {
 			// full config provided, discard default
-			c.DefaultConfig = config
+			//
+			// must be a clone of the default config or the config will be
```

Comment:
1. Maybe to merge these two parts of the comments into one?
2. Is it a clone of the default or of the given config? What I understand is that it is a clone of the given config 🤔 

If my assumptions are correct I would structure the comment as below:

```
full config provided, discard default. It must be a clone of the given(?) config otherwise it will be updated...
```

Reply:
@ChrsMark you are correct, it is the given config.

I updated the comment to be similar to your suggestion